File: view/newmain/theme_classic/view/btns/mainactaprilfoolbtn.py
local var_0_0 = class("MainActAprilFoolBtn", import(".MainBaseActivityBtn"))

def var_0_0.GetEventName(arg_1_0):
	return "event_aprilFool"

def var_0_0.OnInit(arg_2_0):
	local var_2_0 = arg_2_0.IsShowTip()

	setActive(arg_2_0.tipTr.gameObject, var_2_0)

def var_0_0.GetActivityID(arg_3_0):
	return arg_3_0.GetLinkConfig().time[2]

def var_0_0.IsShowTip(arg_4_0):
	local var_4_0 = arg_4_0.GetActivityID()
	local var_4_1 = var_4_0 and getProxy(ActivityProxy).getActivityById(var_4_0)

	return var_4_1 and var_4_1.readyToAchieve()

def var_0_0.CustomOnClick(arg_5_0):
	pg.m02.sendNotification(GAME.GO_SCENE, SCENE.ACTIVITY, {
		id = arg_5_0.GetActivityID()
	})

return var_0_0

File: view/newmain/theme_classic/view/btns/mainactaprilfoolbtn.lua
local var_0_0 = class("MainActAprilFoolBtn", import(".MainBaseActivityBtn"))

function var_0_0.GetEventName(arg_1_0)
	return "event_aprilFool"
end

function var_0_0.OnInit(arg_2_0)
	local var_2_0 = arg_2_0:IsShowTip()

	setActive(arg_2_0.tipTr.gameObject, var_2_0)
end

function var_0_0.GetActivityID(arg_3_0)
	return arg_3_0:GetLinkConfig().time[2]
end

function var_0_0.IsShowTip(arg_4_0)
	local var_4_0 = arg_4_0:GetActivityID()
	local var_4_1 = var_4_0 and getProxy(ActivityProxy):getActivityById(var_4_0)

	return var_4_1 and var_4_1:readyToAchieve()
end

function var_0_0.CustomOnClick(arg_5_0)
	pg.m02:sendNotification(GAME.GO_SCENE, SCENE.ACTIVITY, {
		id = arg_5_0:GetActivityID()
	})
end

return var_0_0
